File: DXGame/DXGame/Models/Entities/Player.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace DXGame.Models.Entities
{
    public class Player
    {
        [Key]
        public string Name { get; set; }
        public ICollection<Playroom> Playrooms { get; set; }
    }
}

File: src/DXGame.Web/src/app/player/player.ts
import { Playroom } from '../playroom/playroom';
import { Guid } from "../common/guid";

export class Player {
    public id: string;
    public name: string;
    public playrooms: Playroom[];
    public activePlayroom: Playroom;
    public protectedWithPassword: boolean;

    constructor(name: string, protectedWithPassword: boolean) {
        this.id = Guid.newGuid();
        this.name = name;
        this.playrooms = [];
        this.protectedWithPassword = protectedWithPassword;
    }
}
